### PR TITLE
Fixed for 1.11, 100% working

### PR DESCRIPTION
**Pull request by Dalethium**
a few fixes that allow this mod to work in 1.11.2

### CODE REVIEW

File: src/main/java/vazkii/ambience/BiomeMapper.java [hunk 1]
```diff
@@ -15,15 +15,11 @@
...
-				biomeMap.put(biome.getBiomeName(), biome);
-		});
-		
-		for(BiomeDictionary.Type t : BiomeDictionary.Type.class.getEnumConstants())
-			typeMap.put(t.getName(), t);
+		 for (ResourceLocation biomeResource : Biome.REGISTRY.getKeys())
+			         {
+			             Biome biome = Biome.REGISTRY.getObject(biomeResource);
+			             biomeMap.put(biome.getBiomeName(), biome);
+			         }
```
Comment (liam4) [inline src/main/java/vazkii/ambience/BiomeMapper.java]: Indentation looks a bit messed up here. Judging by the looks of it the whole repo is done with 8-tabs..

Comment (Dalethium) [inline src/main/java/vazkii/ambience/BiomeMapper.java]: there you go, sorry.

**Comment by Dalethium** [comment]
I found a error and created a fix for it, testing it now.
Stacktrace:
	at net.minecraftforge.fml.common.eventhandler.Event.setCanceled(Event.java:93)
	at vazkii.ambience.Ambience.onBackgroundMusic(Ambience.java:160)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_11_Ambience_onBackgroundMusic_PlaySoundEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
	at net.minecraftforge.client.ForgeHooksClient.playSound(ForgeHooksClient.java:367)
	at net.minecraft.client.audio.SoundManager.func_148611_c(SoundManager.java:344)
	at net.minecraft.client.audio.SoundHandler.func_147682_a(SourceFile:195)
	at net.minecraft.client.audio.MusicTicker.func_181558_a(SourceFile:49)
	at net.minecraft.client.gui.GuiWinGame.func_73876_c(SourceFile:44)

**Comment by Dalethium** [comment]
I fixed the error, I basically muted the music since you can't cancel it I suppose.

File: src/main/java/vazkii/ambience/Ambience.java [hunk 2]
```diff
@@ -157,7 +157,11 @@ public void onRenderOverlay(RenderGameOverlayEvent.Text event) {
 	@SubscribeEvent
 	public void onBackgroundMusic(PlaySoundEvent event) {
 		if(SongLoader.enabled && event.getSound().getCategory() == SoundCategory.MUSIC)
+		if (event.isCancelable()){
```
Comment (liam4) [inline src/main/java/vazkii/ambience/Ambience.java]: Nitpicker here again :) – looks like the rest of the repo uses `if(` vs. `if (`, and `(...) {` vs. `(...){`.

**Comment by Dalethium** [comment]
@liam4 better?

File: src/main/java/vazkii/ambience/Ambience.java [hunk 3]
```diff
@@ -156,11 +156,12 @@ public void onRenderOverlay(RenderGameOverlayEvent.Text event) {
...
-		if (event.isCancelable()){
-			event.setCanceled(true);
-		} else {
-			event.setResultSound(null);
+		if(SongLoader.enabled && event.getSound().getCategory() == SoundCategory.MUSIC) {
+			if(event.isCancelable()) {
+				event.setCanceled(true);
+			} else {
+				event.setResultSound(null);
+			}
```
Comment (liam4) [inline src/main/java/vazkii/ambience/Ambience.java]: Refactoring :+1:

Overall 1.11 support is great! :tada:

**Comment by CircuitLord** [comment]
Could you pre-compile the mod for me? I just tried to compile it myself, but as I haven't done it in a while, I'm not exactly sure where the gradlew.bat build command is outputting to.

**Comment by Vazkii** [comment]
👍

**Comment by CircuitLord** [comment]
That works too :D

**Comment by CircuitLord** [comment]
Lol, pretty much right after he merged them, I realised where gradlew was putting the compiled mod.

**Pull request merged**